plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java

```diff
@@ -954,6 +959,23 @@ public ListObjectsV2Result listObjectsV2(ListObjectsV2Request listObjectsV2Reque
         }
     }
 
+    @Test
+    public void testThatTrinoS3FileSystemReportsConsumedMemory()
+            throws IOException
+    {
+        TestMemoryReservationHandler memoryReservationHandler = new TestMemoryReservationHandler();
+        AggregatedMemoryContext memoryContext = newRootAggregatedMemoryContext(memoryReservationHandler, 1024 * 1000 * 1000);
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            MockAmazonS3 s3 = new MockAmazonS3();
+            Path rootPath = new Path("s3n://test-bucket/");
+            fs.initialize(rootPath.toUri(), newEmptyConfiguration());
+            fs.setS3Client(s3);
+            OutputStream outputStream = fs.create(new Path("s3n://test-bucket/test1"), memoryContext);
+            outputStream.write(new byte[] {1, 2, 3, 4, 5, 6}, 0, 6);
+        }
+        assertThat(memoryReservationHandler.getReserved()).isGreaterThanOrEqualTo(64);
```

Comment:
why `reserved` is not 0? Mem leak?

Reply:
Not a memory leak. in the test I simply don't close outputStream, memory context is not closed and thus not set to 0.
I did it on purpose to be able to catche the value. But I can do it better.